webapp/scripts/query.js

```diff
@@ -21,16 +24,27 @@ class ModuleTwo {
       return messages["7"]
     }
 
-    initMapset('module_2')
+    initMapset(this.mapset)
 
     // set selection as the query area
-    grass('module_2', `g.copy vector=selection@PERMANENT,selection --overwrite`)
+    grass(this.mapset, `g.copy vector=selection@PERMANENT,selection --overwrite`)
     this.queryArea = 'selection'
 
+    const allVector = listUserVector()
+    // check and add all layers from layer switcher
+    getAllFile(CONTAINER_GEOSERVER, [], 'gpkg').forEach(mapFile => {
+      if (allVector.indexOf(mapFile.slice(mapFile.lastIndexOf('/') + 1, mapFile.lastIndexOf('.'))) < 0) {
+        try {
+          addVector('PERMANENT', mapFile, mapFile.slice(mapFile.lastIndexOf('/') + 1, mapFile.lastIndexOf('.')))
+        } catch (e) {
+          console.error(`failed to import ${mapFile} due to error: ${e}`)
```

Comment:
see comments on catch blocks above

Reply:
This one is also a data problem. A number of layers are not importable into GRASS.  But I agree that we should solve it from the root. I will file this issue on GitHub and ask TItusz and Ariel to help.

Reply:
Like the above, I don't think that the situation is resolved by silencing errors. 
(By the way, as GRASS is quite verbose on its own, it's redundant to print the error again. You can leave the catch block empty as well if you want it to be quiet.)